cmd/mtaBuild.go

```diff
@@ -0,0 +1,411 @@
+		if err != nil {
+			return err
+		}
+
+		if err = materialize(config.GlobalSettingsFile, globalSettingsFileDest); err != nil {
+			return err
+		}
+	} else {
+
+		log.Entry().Debugf("Global settings file not provided via configuation.")
+	}
+
+	if len(config.DefaultNpmRegistry) > 0 {
+		log.Entry().Debugf("Setting default npm registry to \"%s\"", config.DefaultNpmRegistry)
+		if err := e.RunExecutable("npm", "config", "set", "registry", config.DefaultNpmRegistry); err != nil {
+			return err
+		}
+	} else {
+		log.Entry().Debugf("No default npm registry provided via configuration. Leaving npm config untouched.")
+	}
+
+	mtaYamlFile := "mta.yaml"
+	mtaYamlFileExists, err := piperutils.FileExists(mtaYamlFile)
+
+	if err != nil {
+		return err
+	}
+
+	if !mtaYamlFileExists {
+
+		log.Entry().Debugf("mta yaml file not found in project sources.")
+
+		if len(config.ApplicationName) == 0 {
+			return fmt.Errorf("'%[1]s' not found in project sources and 'applicationName' not provided as parameter - cannot generate '%[1]s' file", mtaYamlFile)
+		}
+
+		packageFileExists, err := piperutils.FileExists("package.json")
+		if !packageFileExists {
+			return fmt.Errorf("package.json file does not exist")
+		}
+
+		var result map[string]interface{}
+		p, err := ioutil.ReadFile("package.json")
+		if err != nil {
+			return err
+		}
+		json.Unmarshal(p, &result)
+
+		version, ok := result["version"].(string)
+		if !ok {
+			fmt.Errorf("Version not found in \"package.json\" (or wrong type)")
+		}
+
+		name, ok := result["name"].(string)
+		if !ok {
+			fmt.Errorf("Name not found in \"package.json\" (or wrong type)")
+		}
+
+		mtaConfig, err := generateMta(name, config.ApplicationName, version)
+		if err != nil {
+			return err
+		}
+
+		// todo prepare for mocking
+		ioutil.WriteFile(mtaYamlFile, []byte(mtaConfig), 0644)
+		log.Entry().Infof("\"%s\" created.", mtaYamlFile)
+
+	} else {
+		log.Entry().Infof("\"%s\" file found in project sources", mtaYamlFile)
+	}
+
+	mtaYaml, err := ioutil.ReadFile(mtaYamlFile)
+	if err != nil {
+		return err
+	}
+
+	t := time.Now()
+	timestamp := fmt.Sprintf("%d%02d%02d%02d%02d%02d\n", t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second())
+	mtaYamlStr := string(mtaYaml)
+	mtaYamlTimestampReplaced := strings.ReplaceAll(mtaYamlStr, "${timestamp}", timestamp)
+
+	if strings.Compare(mtaYamlStr, mtaYamlTimestampReplaced) != 0 {
+		if err := ioutil.WriteFile(mtaYamlFile, []byte(mtaYamlTimestampReplaced), 0664); err != nil {
+			return err
+		}
+		log.Entry().Debugf("Timestamp replaced in \"%s\"", mtaYamlFile)
+	}
+
+	var call []string
+
+	mtarName := config.MtarName
+	if len(mtarName) == 0 {
+		log.Entry().Debugf("mtar name not provided via config. Extracting from file \"%s\"", mtaYamlFile)
+		mtaID, err := getMtaID(mtaYamlFile)
+		if err != nil {
+			return err
+		}
+		log.Entry().Debugf("mtar name extracted from file \"%s\": \"%s\"", mtaYamlFile, mtaID)
+		mtarName = mtaID + ".mtar"
+	}
+
+	var mtaJar = "mta.jar"
```

Comment:
please inline into the `call`

Reply:
@CCFenner . We cannot inline this since there is a parameter providing the jar file location from outside. Was missing here, but has been added in the meantime (wip ...).